debezium-server/debezium-server-http/src/main/java/io/debezium/server/http/HttpChangeConsumer.java

```diff
@@ -0,0 +1,130 @@
+
+    private static String sinkUrl;
+    private static Duration timeoutDuration;
+    private static String contentType;
+
+    private HttpClient client;
+    private HttpRequest.Builder requestBuilder;
+
+    // If this is running as a Knative object, then expect the sink URL to be located in `K_SINK`
+    // as per https://knative.dev/development/eventing/custom-event-source/sinkbinding/
+    @PostConstruct
+    void connect() throws URISyntaxException {
+        client = HttpClient.newHttpClient();
+        final Config config = ConfigProvider.getConfig();
+        String sink = System.getenv("K_SINK");
+        timeoutDuration = Duration.ofMillis(HTTP_TIMEOUT);
+
+        if (sink != null) {
+            sinkUrl = sink;
+        }
+        else {
+            sinkUrl = config.getValue(PROP_PREFIX + PROP_WEBHOOK_URL, String.class);
+        }
+
+        config.getOptionalValue(PROP_PREFIX + PROP_CLIENT_TIMEOUT, String.class)
+                .ifPresent(t -> timeoutDuration = Duration.ofMillis(Long.parseLong(t)));
+
+        switch (config.getValue("debezium.format.value", String.class)) {
+            case "avro":
+                contentType = "avro/bytes";
+                break;
+            case "cloudevents":
+                contentType = "application/cloudevents+json";
+                break;
+            default:
+                // Note: will default to JSON if it cannot be determined, but should not reach this point
+                contentType = "application/json";
+        }
+
+        LOGGER.info("Using http content-type type {}", contentType);
+        LOGGER.info("Using sink URL: {}", sinkUrl);
+        requestBuilder = HttpRequest.newBuilder(new URI(sinkUrl)).timeout(timeoutDuration);
+        requestBuilder.setHeader("content-type", contentType);
+    }
+
+    @Override
+    public void handleBatch(List<ChangeEvent<Object, Object>> records, DebeziumEngine.RecordCommitter<ChangeEvent<Object, Object>> committer)
+            throws InterruptedException {
+        for (ChangeEvent<Object, Object> record : records) {
+            LOGGER.trace("Received event '{}'", record);
+            String value = "";
+            HttpResponse r;
+
+            if (record.value() != null) {
+                value = (String) record.value();
+
+                HttpRequest request = requestBuilder.POST(HttpRequest.BodyPublishers.ofString(value)).build();
+
+                try {
+                    r = client.send(request, HttpResponse.BodyHandlers.ofString());
+                }
+                catch (IOException ioe) {
+                    throw new InterruptedException(ioe.toString());
+                }
+
+                if ((r.statusCode() == HTTP_OK) || (r.statusCode() == HTTP_NO_CONTENT) || (r.statusCode() == HTTP_ACCEPTED)) {
+                    try {
+                        committer.markProcessed(record);
+                    }
+                    catch (InterruptedException e) {
```

Comment:
Just realized this try/catch block is no longer necessary and will be removing it.